server/services/CsvGeneratorService v2.js

```diff
@@ -0,0 +1,787 @@
+    try {
+      const { callWithRequest } = this.esDriver.getCluster('data');
+      const savedsearchIdFromES = await callWithRequest(_req, 'get', {
+        index: '.kibana',
+        id: 'search:' + savedsearchId,
+      });
+      return savedsearchIdFromES;
+    } catch (err) {
+      console.error(
+        'getSavedSearch Infos - Error while fetching savedSearch Infos from elasticsearch:',
+        err
+      );
+      return { ok: false, resp: err.message };
+    }
+  };
+  indexPattern = async (_req, indexpatternId) => {
+    const { callWithRequest }  = this.esDriver.getCluster('data');
+    const indexpatern          = await callWithRequest(_req, 'get', {
+      index: '.kibana',
+      id: 'index-pattern:' + indexpatternId,
+    });
+    return indexpatern;
+  };
+  esFetchCount = async (_req, indexPatternTitle, bodyCount) => {
+    try {
+      const { callWithRequest } = this.esDriver.getCluster('data');
+      const fecthCountRes       = callWithRequest(_req, 'count', {
+        index: indexPatternTitle,
+        body: bodyCount,
+      });
+      return fecthCountRes;
+    } catch (err) {
+      console.error(
+        'CSV Generator - CsvGeneratorService - ESFetchCount - Error while counting the number of elements in ElasticSearch ',
+        err
+      );
+      return { ok: false, resp: err.message };
+    }
+  };
+  esFetchData = async (_req, indexPatternTitle, body) => {
+    try {
+      const { callWithRequest } = this.esDriver.getCluster('data');
+      const fecthDataRes        = callWithRequest(_req, 'search', {
+        index: indexPatternTitle,
+        scroll: '1m',
+        body: body,
+      });
+      return fecthDataRes;
+    } catch (err) {
+      console.error(
+        'CSV Generator - CsvGeneratorService - ESFetchData - Error while Fetching the data from ElasticSearch ',
+        err
+      );
+      return { ok: false, resp: err.message };
+    }
+  };
+  esFetchScroll = async (_req, scrollId) => {
+    try {
+      const { callWithRequest } = this.esDriver.getCluster('data');
+      const fecthDataScrollRes  = callWithRequest(_req, 'scroll', {
+        scrollId: scrollId,
+        scroll: '1m',
+      });
+      return fecthDataScrollRes;
+    } catch (err) {
+      console.error(
+        'CSV Generator - CsvGeneratorService - ESFetchScroll - Error while Fetching the scroll data from ElasticSearch ',
+        err
+      );
+      return { ok: false, resp: err.message };
+    }
+  };
+  traverse = (data, keys, result = {}) => {
+    for (const k of Object.keys(data)) {
+      if (keys.includes(k)) {
+        result = {
+          ...result,
+          ...{
+            [k]: data[k],
+          },
+        };
+        continue;
+      }
+      if (data[k] && typeof data[k] === 'object' && Object.keys(data[k]).length > 0) {
+        result = this.traverse(data[k], keys, result);
+      }
+    }
+    return result;
+  };
+
+  genereteCsv = async (_req, documentId, username, strFilename) => {
```

Comment:
try split these almost 400 lines big method to small methods, for readability, maintainability , testability

try control length of method(no more than 100 lines per each method), extra utility method